Universidad Torcuato Di Tella

Licenciatura en Tecnología Digital\
**Tecnología Digital VI: Inteligencia Artificial**

# **Trabajo Práctico N°1**

El objetivo de este trabajo práctico es realizar un análisis completo del ciclo de vida de un problema de Machine Learning, centrándose en la preparación de los datos, la prevención de Data Leakage, la estrategia de validación y la correcta elección de métricas orientadas al negocio.

A continuación, importamos los módulos correspondientes.

In [5]:
# Importamos
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
# Importamos el dataset de Telco Customer Churn
dataset:pd.DataFrame = pd.read_csv("dataset/WA_Fn-UseC_-Telco-Customer-Churn.csv")

# Imprimimos las primeras cinco filas
dataset.head()

Dimensiones del dataset: (7043, 21)


,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


### **Descripción concisa del conjunto de datos y el problema de negocio a resolver**

Los datos provistos nos hablan acerca de un registro de clientes de la empresa Telco. Cada fila representa un usuario junto con una serie de características, entre las cuales se encuentra el atributo de `churn`, que nos indica si dicho cliente abandono o no el servicio en el último mes.

El problema a resolver sobre el cual trabajaremos consiste en predecir si un usuario esta o no por abandonar el servicio (*Churn Prediction*). Esta clasificación se logra a partir de distinguir qué características predominan entre los usuarios que abandonaron.

### **Análisis exploratorio**

Procedemos con un análisis exhaustivo del conjunto de datos para entender como se comportan.

In [ ]:
# Dimensiones del dataset
dataset.shape

(7043, 21)

Podemos observar que el dataset que disponemos cuenta con **7043 muestras** y cada una cuenta con **21 features**.

In [25]:
# Aislamos la columna `churn` para ver como se distribuyen las clases (Yes/No)
churn:pd.DataFrame = pd.DataFrame(dataset['Churn'])

Notamos un **gran desbalance entre las clases `Yes` y `No`en la feature `Churn`** ya que la mayoria de los usuarios recolectados para el dataset les corresponde la clase `No`, es decir, son usuarios que no abandonaron el servicio en el último mes. Luego, `No` es la clase mayoritaría y `Yes` la clase minoritaria.

Para explícitar este desbalance vamos a proceder con un bar-plot.

In [17]:
churn.columns

Index(['Churn'], dtype='object')